Futures-util dev dependency should require alloc feature

We had the following CI failure on Tokio's run-hyper-tests job:
```
error[E0425]: cannot find function `join_all` in module `future`
   --> tests/support/mod.rs:477:26
    |
477 |         Box::pin(future::join_all(client_futures).map(|_| ()))
    |                          ^^^^^^^^ not found in `future`
    |
note: found an item that was configured out
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.30/src/future/mod.rs:81:5
    |
81  | mod join_all;
    |     ^^^^^^^^
    = note: the item is gated behind the `alloc` feature
note: found an item that was configured out
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.30/src/future/mod.rs:83:26
    |
83  | pub use self::join_all::{join_all, JoinAll};
    |                          ^^^^^^^^
    = note: the item is gated behind the `alloc` feature
```
It seems like the futures-util crate has an alloc feature that should be enabled.

Most likely, it was previously enabled by some other dependency that no longer enables alloc.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,7 +43,7 @@ want = { version = "0.3", optional = true }
 [dev-dependencies]
 form_urlencoded = "1"
 futures-channel = { version = "0.3", features = ["sink"] }
-futures-util = { version = "0.3", default-features = false, features = ["sink"] }
+futures-util = { version = "0.3", default-features = false, features = ["alloc", "sink"] }
 http-body-util = "0.1"
 pretty_env_logger = "0.5"
 spmc = "0.3"
@@ -85,7 +85,7 @@ client = ["dep:want", "dep:pin-project-lite", "dep:smallvec"]
 server = ["dep:httpdate", "dep:pin-project-lite", "dep:smallvec"]
 
 # C-API support (currently unstable (no semver))
-ffi = ["dep:libc", "dep:http-body-util"]
+ffi = ["dep:libc", "dep:http-body-util", "futures-util?/alloc"]
 
 # Utilize tracing (currently unstable)
 tracing = ["dep:tracing"]

